### PR TITLE
session-desktop-appimage: init at 1.7.1

### PR DESCRIPTION
**Pull request by Alexnortung**
###### Motivation for this change

Adding the instant messenger session-desktop to the nix package manager

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -450,6 +450,12 @@
     githubId = 2335822;
     name = "Alexandre Esteves";
   };
+  alexnortung = {
+    name = "alexnortung";
+    email = "alex_nortung@live.dk ";
```
Comment (figsoda) [inline maintainers/maintainer-list.nix:455]: This should also be in a separate commit: `maintainers: add alexnortung`
```suggestion
    email = "alex_nortung@live.dk";
```

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+, appimageTools
+}:
+
+let
+  version = "1.6.11";
+in
+appimageTools.wrapType2 {
+  name = "session-desktop-appimage";
+  src = fetchurl {
+    url = "https://github.com/oxen-io/session-desktop/releases/download/v${version}/session-desktop-linux-x86_64-${version}.AppImage";
```
Comment (figsoda) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:12]:
```suggestion
appimageTools.wrapType2 rec {
  pname = "session-desktop-appimage";
  version = "1.6.11";

  src = fetchurl {
    url = "https://github.com/oxen-io/session-desktop/releases/download/v${version}/session-desktop-linux-x86_64-${version}.AppImage";
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ lib
+, fetchurl
+, appimageTools
+}:
+
+let
+  version = "1.6.11";
+in
+appimageTools.wrapType2 {
+  name = "session-desktop-appimage";
```
Comment (figsoda) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:10]:
```suggestion
  name = "session-desktop-appimage-${version}";
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchurl {
+    url = "https://github.com/oxen-io/session-desktop/releases/download/v${version}/session-desktop-linux-x86_64-${version}.AppImage";
+    sha256 = "08vnkxj2w2k7lhygs5qhbyikpvw5nmby3qhxigbbkj5qj1wvs48b";
+  };
+
+  meta = with lib; {
+    description = "Onion routing based messenger";
+    homepage = "https://getsession.org/";
+    license = licenses.gpl3;
```
Comment (figsoda) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:19]: specify `gpl3Only` or `gpl3Plus`

File: pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    maintainers = with maintainers; [ alexnortung ];
+    platforms = platforms.linux;
```
Comment (figsoda) [inline pkgs/applications/networking/instant-messengers/session-desktop-appimage/default.nix:21]: doesn't seem to work on aarch64-linux

```suggestion
    platforms = [ "x86_64-linux" ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]

**Review by figsoda — APPROVED** [review]
lgtm

**Comment by Alexnortung** [comment]
Wait I forgot to add it to all packages again. I will do it later

**Comment by Alexnortung** [comment]
I believe it is ready now

**Comment by figsoda** [comment]
1.7.1 is out. do you mind making an update?

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 135831` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>session-desktop-appimage</li>
  </ul>
</details>
